### ISSUE TITLE
Can't build flatpak that depends on mpmath

### ISSUE THREAD

**Issue by TabulateJarl8** [flatpak/flatpak-builder-tools#232]
Hey, I was working on build a flatpak again today after my poetry issue was resolved, and I came across another issue. I don't seem to be able to build stuff when mpmath is in my requirements file. I can manually add it and it works just fine, but it doesn't work in the auto-generated file. Here's the error:

```py
========================================================================
Building module python3-mpmath in /home/tabulate/TabulateCalc/.flatpak-builder/build/python3-mpmath-1
========================================================================
Running: pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "mpmath>=1.0.0" --no-build-isolation
Using pip 21.0.1 from /usr/lib/python3.8/site-packages/pip (python 3.8)
Non-user install due to --prefix or --target option
Ignoring indexes: https://pypi.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-k4gxpgzn
Created temporary directory: /tmp/pip-req-tracker-96_ai2fm
Initialized build tracking at /tmp/pip-req-tracker-96_ai2fm
Created build tracker: /tmp/pip-req-tracker-96_ai2fm
Entered build tracker: /tmp/pip-req-tracker-96_ai2fm
Created temporary directory: /tmp/pip-install-kid0hmk2
Looking in links: file:///run/build/python3-mpmath
0 location(s) to search for versions of mpmath:
Skipping link: not a file: file:///run/build/python3-mpmath (from -f)
Found link file:///run/build/python3-mpmath/mpmath-1.2.1.tar.gz, version: 1.2.1
Local files found: /run/build/python3-mpmath/mpmath-1.2.1.tar.gz
Given no hashes to check 1 links for project 'mpmath': discarding no candidates
Processing ./mpmath-1.2.1.tar.gz
  Added mpmath>=1.0.0 from file:///run/build/python3-mpmath/mpmath-1.2.1.tar.gz to build tracker '/tmp/pip-req-tracker-96_ai2fm'
    Running setup.py (path:/tmp/pip-install-kid0hmk2/mpmath_19997ce3fea3434cb8b3482677922993/setup.py) egg_info for package mpmath
    Created temporary directory: /tmp/pip-pip-egg-info-dqwymklz
    Running command python setup.py egg_info
    WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4dec822b50>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools-scm/
    WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4dec822d60>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools-scm/
    WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4dec822f10>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools-scm/
    WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4dec7cd100>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools-scm/
    WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4dec7cd2b0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools-scm/
    ERROR: Could not find a version that satisfies the requirement setuptools_scm>=1.7.0
    ERROR: No matching distribution found for setuptools_scm>=1.7.0
    Traceback (most recent call last):
      File "/usr/lib/python3.8/site-packages/setuptools/installer.py", line 119, in fetch_build_egg
        subprocess.check_call(cmd)
      File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp2kc02875', '--quiet', 'setuptools_scm>=1.7.0']' returned non-zero exit status 1.

    The above exception was the direct cause of the following exception:

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-kid0hmk2/mpmath_19997ce3fea3434cb8b3482677922993/setup.py", line 6, in <module>
        setuptools.setup(use_scm_version=True)
      File "/usr/lib/python3.8/site-packages/setuptools/__init__.py", line 152, in setup
        _install_setup_requires(attrs)
      File "/usr/lib/python3.8/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
        dist.fetch_build_eggs(dist.setup_requires)
      File "/usr/lib/python3.8/site-packages/setuptools/dist.py", line 686, in fetch_build_eggs
        resolved_dists = pkg_resources.working_set.resolve(
      File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 766, in resolve
        dist = best[req.key] = env.best_match(
      File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1051, in best_match
        return self.obtain(req, installer)
      File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1063, in obtain
        return installer(requirement)
      File "/usr/lib/python3.8/site-packages/setuptools/dist.py", line 745, in fetch_build_egg
        return fetch_build_egg(self, req)
      File "/usr/lib/python3.8/site-packages/setuptools/installer.py", line 121, in fetch_build_egg
        raise DistutilsError(str(e)) from e
    distutils.errors.DistutilsError: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp2kc02875', '--quiet', 'setuptools_scm>=1.7.0']' returned non-zero exit status 1.
WARNING: Discarding file:///run/build/python3-mpmath/mpmath-1.2.1.tar.gz. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement mpmath>=1.0.0
ERROR: No matching distribution found for mpmath>=1.0.0
Exception information:
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 171, in _merge_into_criterion
    crit = self.state.criteria[name]
KeyError: 'mpmath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 318, in resolve
    name, crit = self._merge_into_criterion(r, parent=None)
  File "/usr/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 173, in _merge_into_criterion
    crit = Criterion.from_requirement(self._p, requirement, parent)
  File "/usr/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 83, in from_requirement
    raise RequirementsConflicted(criterion)
pip._vendor.resolvelib.resolvers.RequirementsConflicted: Requirements conflict: SpecifierRequirement('mpmath>=1.0.0')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 121, in resolve
    self._result = resolver.resolve(
  File "/usr/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 453, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/usr/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 320, in resolve
    raise ResolutionImpossible(e.criterion.information)
pip._vendor.resolvelib.resolvers.ResolutionImpossible: [RequirementInformation(requirement=SpecifierRequirement('mpmath>=1.0.0'), parent=None)]

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 189, in _main
    status = self.run(options, args)
  File "/usr/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 178, in wrapper
    return func(self, options, args)
  File "/usr/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 316, in run
    requirement_set = resolver.resolve(
  File "/usr/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 127, in resolve
    six.raise_from(error, e)
  File "<string>", line 3, in raise_from
pip._internal.exceptions.DistributionNotFound: No matching distribution found for mpmath>=1.0.0
Removed mpmath>=1.0.0 from file:///run/build/python3-mpmath/mpmath-1.2.1.tar.gz from build tracker '/tmp/pip-req-tracker-96_ai2fm'
Removed build tracker: '/tmp/pip-req-tracker-96_ai2fm'
Error: module python3-mpmath: Child process exited with code 1
```

The name resolution error doesn't make much sense since I can access that URL outside of the build environment, and I seem to be getting a similar KeyError to the one in #226. If I remove mpmath from my requirements and instead add it manually with this code, it works:
```yml
- name: mpmath
  buildsystem: simple
  build-commands:
    - pip3 install --prefix /app --no-deps .
  sources:
    - type: archive
      url: https://github.com/fredrik-johansson/mpmath/archive/refs/tags/1.0.0.tar.gz
      sha256: f4edfddb63c0f87d8ae637290d2fd1743d47aec5452453e50d8d9c3c3fdf57e0
```
Here's the relevant section of my `python3-requirements.json`:
```json
{
            "name": "python3-mpmath",
            "buildsystem": "simple",
            "build-commands": [
                "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"mpmath>=1.0.0\" --no-build-isolation"
            ],
            "sources": [
                {
                    "type": "file",
                    "url": "https://files.pythonhosted.org/packages/95/ba/7384cb4db4ed474d4582944053549e02ec25da630810e4a23454bc9fa617/mpmath-1.2.1.tar.gz",
                    "sha256": "79ffb45cf9f4b101a807595bcb3e72e0396202e0b1d25d689134b48c4216a81a"
                }
            ]
        }
```
Thanks for any help

**Comment by hfiguiere** [comment]
I think this is fixed with #238 if you can retest this solve your issue.

**Comment by TabulateJarl8** [comment]
> I think this is fixed with #238 if you can retest this solve your issue.

Yep, that fixed it. Thanks!
